src/js/components/conditionalLogic.js

```diff
@@ -131,6 +131,13 @@ export default class ConditionalLogic {
                         if (action.fields) {
                             action.fields.forEach((fieldName) => {
                                 let field = this.form.querySelectorAll('[name*="' + fieldName + '"]');
+
+                                // There is also a BE Bug that not all elements are getting the name property.
+                                // This fixes the issue that no conditional logic can be defined for html elements and snippets.
+                                if (!field) {
+                                    field = this.form.querySelectorAll('[data-field-id*="' + fieldName + '"]');
```

Comment:
I would recommend to update the selector on the previous `querySelectorAll` to search for both at once:

```
let field = this.form.querySelectorAll('[name*="' + fieldName + '"], [data-field-id*="' + fieldName + '"]');
```

See also https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll#obtaining_a_list_of_matches